src/main/kotlin/io/curity/identityserver/plugins/authenticator/authentication/CallbackRequestHandler.kt

```diff
@@ -95,6 +95,26 @@ class CallbackRequestHandler(
         return Optional.ofNullable(AuthenticationResult(attributes))
     }
 
+    private fun fetchUserInfoClaims(providerAccessToken:String): JwtClaims{
+        val userInfoResponse = _config.getHttpClient()
+            .request(_providerConfiguration.userInfoEndpoint)
+            .header("Authorization", "Bearer $providerAccessToken")
+            .post()
+            .response()
+
+        return JwtClaims.parse(userInfoResponse.body(HttpResponse.asString()))
+    }
+
+    private fun prepareSubjectAttributes(idTokenClaims:JwtClaims, userInfoClaims:JwtClaims?): SubjectAttributes {
+        val subjectAttributesFromIdToken = idTokenClaims.claimsMap.filter { it.key !in FILTER_CLAIMS }
+            .toMutableMap().apply { this["subject"] = idTokenClaims.subject }
+        val subjectAttributesFromUserInfo = userInfoClaims?.claimsMap?.filter { it.key !in FILTER_CLAIMS }
```

Comment:
Could be good to log both maps masked, in case we have some unexpected results. If the claim values are different, userinfo gets precedence.

Reply:
Fixed.